nsq/reader.py

```diff
@@ -275,6 +283,18 @@ def set_message_handler(self, message_handler):
         assert callable(message_handler), 'message_handler must be callable'
         self.message_handler = message_handler
 
+    def set_message_timeout_handler(self, message_timeout_handler):
+        """
+        Assigns the callback method to be executed for each message timeout that occurs. This method can only be used
+            in conjunction with the client-side msg_timeout setting. The client, otherwise, has no awareness of the
+            message's lifespan.
```

Comment:
`nsqd` delivers the configured `msg_timeout` to the client in the response body of the `IDENTIFY` command, so the client always knows what the `msg_timeout` is.